Parse/Public/Utilities/Conversion.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2015-present, Parse, LLC.  All rights reserved.  This source code is licensed under the BSD-style license found in the LICENSE file in the root directory of this source tree.  An additional grant of patent rights can be found in the PATENTS file in the same directory.
+
+using Parse.Internal;
+using System;
+using System.Collections.Generic;
+
+namespace Parse.Utilities {
+  /// <summary>
+  /// A set of utilities for converting generic types between each other.
+  /// </summary>
+  public static class Conversion {
```

Comment:
I think we want it public for now, if we decide to do it as a separate project/repository, we'll make it a dependency, and it would stay under the same namespace.